pkg/operator/configobservation/cloudprovider/observe_cloudprovider.go

```diff
@@ -26,22 +34,62 @@ func ObserveCloudProviderNames(genericListers configobserver.Listers, recorder e
 		}
 	}
 
+	if currentCloudConfig, _, _ := unstructured.NestedStringSlice(existingConfig, cloudProviderConfPath...); len(currentCloudConfig) > 0 {
+		if err := unstructured.SetNestedStringSlice(previouslyObservedConfig, currentCloudConfig, cloudProviderConfPath...); err != nil {
+			errs = append(errs, err)
+		}
+	}
+
 	observedConfig := map[string]interface{}{}
 
 	infrastructure, err := listers.InfrastructureLister.Get("cluster")
 	if errors.IsNotFound(err) {
-		recorder.Warningf("ObserveRestrictedCIDRFailed", "Required infrastructures.%s/cluster not found", configv1.GroupName)
+		recorder.Warningf("ObserverCloudProviderNames", "Required infrastructures.%s/cluster not found", configv1.GroupName)
 		return observedConfig, errs
 	}
 	if err != nil {
 		return previouslyObservedConfig, errs
 	}
 
+	cloudProvider := getPlatformName(infrastructure.Status.Platform, recorder)
+	if len(cloudProvider) > 0 {
+		if err := unstructured.SetNestedStringSlice(observedConfig, []string{cloudProvider}, cloudProvidersPath...); err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	sourceCloudConfigMap := infrastructure.Spec.CloudConfig.Name
+	sourceCloudConfigNamespace := configNamespace
+	if len(sourceCloudConfigMap) == 0 {
+		return observedConfig, errs
+	}
+
+	err = listers.ResourceSyncer().SyncConfigMap(
+		resourcesynccontroller.ResourceLocation{
+			Namespace: targetNamespaceName,
+			Name:      "cloud-config",
+		},
+		resourcesynccontroller.ResourceLocation{
+			Namespace: sourceCloudConfigNamespace,
+			Name:      sourceCloudConfigMap,
+		},
+	)
+	if err != nil {
+		errs = append(errs, err)
+		return observedConfig, errs
+	}
+	if err := unstructured.SetNestedStringSlice(observedConfig, []string{cloudProviderConfFilePath}, cloudProviderConfPath...); err != nil {
+		recorder.Warningf("ObserverCloudProviderNames", "Failed setting cloud-config : %v", err)
+		errs = append(errs, err)
+	}
+	return observedConfig, errs
```

Comment:
just before returning, can you look for a diff between the prevconfig and the newconfig and emit an event.  @mfojtik probably has an example.

Reply:
^ does this work or did you mean deep comparsion of actual cloudprovider configuration?